packages/nx/src/config/workspaces.ts

```diff
@@ -606,10 +611,10 @@ function resolvePathTokensInOptions<T extends Object | Array<unknown>>(
   const result: T = Array.isArray(object) ? ([...object] as T) : { ...object };
   for (let [opt, value] of Object.entries(object ?? {})) {
     if (typeof value === 'string') {
-      if (value.startsWith('{workspaceRoot}/')) {
-        value = value.replace(/^\{workspaceRoot\}\//, '');
-      }
-      if (value.includes('{workspaceRoot}')) {
+      const workspaceRootMatch = /^(\{workspaceRoot\}\/?)/.exec(value);
+      if (workspaceRootMatch?.length) {
+        value = value.replace(workspaceRootMatch[0], '');
+      } else if (value.includes('{workspaceRoot}')) {
```

Comment:
The handling for this changed a bit as I noticed a potential bug when doing it. If an option represented a path to a directory, `{workspaceRoot}` may be a valid value for that, and this would throw as before we required it to have a `/` after it.